### PR TITLE
Cleaning WMTS utils code [Server]

### PR DESCRIPTION
**Pull request by rldhont**
## Description

In the tile matrices calculation the code could be simplified.

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nirvn** [comment]
In theory, does WMTS not allow for non square tiles?

File: src/server/services/wmts/qgswmtsutils.cpp [hunk 1]
```diff
@@ -34,15 +34,12 @@ namespace QgsWmts
...
     QMap< QString, tileMatrixInfo> populateTileMatrixInfoMap();
 
     QgsCoordinateReferenceSystem wgs84 = QgsCoordinateReferenceSystem::fromOgcWmsCrs( GEO_EPSG_CRS_AUTHID );
 
-    double METERS_PER_INCH = 0.02540005080010160020;
-    QMap< QgsUnitTypes::DistanceUnit, double> INCHES_PER_UNIT = populateInchesPerUnit();
-    int tileWidth = 256;
-    int tileHeight = 256;
+    int tileSize = 256;
+    double POINTS_TO_M = 2.83464567 / 10000.0;
```
Comment (pblottiere) [inline src/server/services/wmts/qgswmtsutils.cpp]: Why `POINTS_TO_M` instead of `points_to_m`?

File: src/server/services/wmts/qgswmtsutils.cpp [hunk 2]
```diff
@@ -102,18 +99,19 @@ namespace QgsWmts
 
     // calculate tile matrix scale denominator
     double scaleDenominator = 0.0;
-    int colRes = ( tmi.extent.xMaximum() - tmi.extent.xMinimum() ) / tileWidth;
-    int rowRes = ( tmi.extent.yMaximum() - tmi.extent.yMinimum() ) / tileHeight;
+    int colRes = ( tmi.extent.xMaximum() - tmi.extent.xMinimum() ) / tileSize;
+    int rowRes = ( tmi.extent.yMaximum() - tmi.extent.yMinimum() ) / tileSize;
+    double UNIT_TO_M = QgsUnitTypes::fromUnitToUnitFactor( tmi.unit, QgsUnitTypes::DistanceMeters );
```
Comment (pblottiere) [inline src/server/services/wmts/qgswmtsutils.cpp]: `unit_to_m`

File: src/server/services/wmts/qgswmtsutils.cpp [hunk 3]
```diff
@@ -152,25 +150,33 @@ namespace QgsWmts
     QgsRectangle extent = tmi.extent;
     QgsUnitTypes::DistanceUnit unit = tmi.unit;
 
+    // constant
+    double UNIT_TO_M = QgsUnitTypes::fromUnitToUnitFactor( tmi.unit, QgsUnitTypes::DistanceMeters );
```
Comment (pblottiere) [inline src/server/services/wmts/qgswmtsutils.cpp]: `unit_to_m`

File: src/server/services/wmts/qgswmtsutils.cpp (hunk 1 above)
Comment (elpaso) [reply src/server/services/wmts/qgswmtsutils.cpp]: Should it be `constexpr` ? (or at least `const`)

**Comment by rldhont** [comment]
@pblottiere can you review the code ?

**Pull request merged**